app/src/main/scala/com/waz/zclient/conversationlist/ConversationListController.scala

```diff
@@ -219,6 +220,23 @@ class ConversationListController(implicit inj: Injector, ec: EventContext)
     case ex: Exception => error(l"error while creating custom folder $folderName for conv $convId", ex)
       Future.failed(ex)
   }
+
+  def deleteConversation(teamId: Option[TeamId], convId: ConvId): Future[Unit] = teamId match {
+    case None    => Future.successful(())
+    case Some(tid) =>
+      (for {
+        contUpdater <- convService.head.map(_.content)
+        convOpt     <- contUpdater.convById(convId)
+        Some(conv)   = convOpt
```

Comment:
I know I taught you this, but as Maciek pointed out, it's not good practice.

Reply:
it gets too long when I use `match` statement. the code evolved somehow to nested for comprehensions which looked strange. I couldn't see any way out of it so cheated a bit w/ using flatMap to make it more readable. I tried to simplify it w/ `collect` too, but let me know if that needs change.

Reply:
I think you could do something like...

```suggestion
for {
  contUpdater <- convService.head.map(_.content)
  convOpt <- contUpdater.convById(convId)
  teamService <- inject[Signal[TeamsService]].head 
} yield convOpt match {
  case Some(conv) => teamService.deleteGroupConversation(did, conv.remoteId)
  case None => Future.successful(())
}
```